src/nix/diff-closures.cc

```diff
@@ -91,13 +103,42 @@ void printClosureDiff(
         for (auto & [version, _] : afterVersions)
             if (!beforeVersions.count(version)) added.insert(version);
 
-        if (showDelta || !removed.empty() || !added.empty()) {
-            std::vector<std::string> items;
-            if (!removed.empty() || !added.empty())
-                items.push_back(fmt("%s → %s", showVersions(removed), showVersions(added)));
+        if (!removed.empty() || !added.empty()) {
+            auto info = DiffInfoForPackage {
+                .sizeDelta = sizeDelta,
+                .addedVersions = showVersions(added),
+                .removedVersions = showVersions(removed)
+            };
+            itemsToPrint[name] = std::move(info);
+        }
+    }
```

Comment:
Can we return `itemsToPrint`, and then have the rendering in a separate function? I think that is a bit cleaner, and thanks to the fact that you created the type it is now possible to do so! :)